pkg/chain/ethereum/bitcoin_difficulty.go

```diff
@@ -0,0 +1,102 @@
+package ethereum
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/keep-network/keep-common/pkg/chain/ethereum"
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/chain"
+	"github.com/keep-network/keep-core/pkg/chain/ethereum/tbtc/gen/contract"
+)
+
+// Definitions of contract names.
+const (
+	LightRelayContractName = "LightRelay"
+)
+
+// BitcoinDifficultyChain represents a Bitcoin difficulty-specific chain handle.
+type BitcoinDifficultyChain struct {
+	*baseChain
+
+	lightRelay *contract.LightRelay
+}
+
+// NewBitcoinDifficultyChain construct a new instance of the Bitcoin difficulty
+// - specific Ethereum chain handle.
+func NewBitcoinDifficultyChain(
+	config ethereum.Config,
+	baseChain *baseChain,
+) (*BitcoinDifficultyChain, error) {
+	lightRelayAddress, err := config.ContractAddress(LightRelayContractName)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to attach to LightRelay contract: [%v]",
+			err,
+		)
+	}
+
+	lightRelay, err :=
+		contract.NewLightRelay(
+			lightRelayAddress,
+			baseChain.chainID,
+			baseChain.key,
+			baseChain.client,
+			baseChain.nonceManager,
+			baseChain.miningWaiter,
+			baseChain.blockCounter,
+			baseChain.transactionMutex,
+		)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to attach to EcdsaSortitionPool contract: [%v]",
```

Comment:
```suggestion
			"failed to attach to LightRelay contract: [%v]",
```

Reply:
Done in c3bac43be0723d973ae2d095979f3e105945fb35.